students/anthony_mckeever/lesson5/assignment_1/mailroom.py

```diff
@@ -0,0 +1,397 @@
+    """
+    output = None
+    try:
+        output = input(msg)
+    except (KeyboardInterrupt, EOFError):
+        print("Exiting...")
+        sys.exit()
+    else:
+        return output
+
+
+def menu_system(opts_dict, menu_text, prompt_text, include_main=False, include_donors=False, invalid_opt=None):
+    """
+    The menu system that handles most user multiple choice inputs.
+
+    :opts_dict:         The initial options dictionary.
+    :menu_text:         The flavor text of the menu.
+    :prompt_text:       What to prompt the user when accepting input.
+    :include_main:      Whether or not to include options for returning to the main menu (Default = False)
+    :include_donors:    Whether or not to include the donors with the options. (Default = False)
+                        Note: Donors will not print when menu options are printed.
+    :invalid_opt:       Override the action if the user provides an invalid selection. (Default = None)
+                        Note: Using None will prompt user to reattemt their choice until a correct choice is made.
+    """
+    show_opts = opts_dict.copy()
+    
+    if include_main:
+        show_opts.update(main_dict)
+
+    show_opts.update(quit_dict)
+
+    print(menu_text)
+    print_options(show_opts)
+    
+    # Don't include the list of donors in the help text so the user has to request them.
+    if include_donors:
+        show_opts.update(donor_dict)
+
+    while True:
+        user_choice = safe_input(prompt_text)
+        choice_key = key_from_lower(user_choice, show_opts.keys())
+
+        if choice_key is not None:
+            selection = show_opts[choice_key]
+
+            if selection[1]:
+                if selection[1] == accept_donation:
+                    selection[1](choice_key)
+                else:
+                    selection[1]()
+            
+            if selection[1] != print_donors:
+                break
+
+        else:
+            if invalid_opt is None:
+                print("Invalid choice.  Please select from available options.")
+            else:
+                invalid_opt(user_choice)
+                break
+
+
+def print_options(show_opts):
+    """
+    Print the options from a menu.
+
+    :show_opts:     The complete dictionary of the current menu's options.
+    """
+    skip = set([])
+    for key, value in show_opts.items():
+        if key not in skip:
+            opts_string, skippable = get_opts_string(key, show_opts)
+            print("\t" + opts_string + value[0])
+            skip.update(skippable)
+            
+
+def key_from_lower(user_choice, keys):
+    """
+    Return the user's lower case input to a properly cased dictionary key.
+
+    :user_choice:   The choice the user made at a menu.
+    :keys:          The keys from the dictionary the user is choosing from.
+    """
+    lower_keys = {k.lower(): k for k in keys}
+    return lower_keys[user_choice] if user_choice in lower_keys.keys() else None
```

Comment:
I don't really understand the purpose of this function, however this can be simplified to 
`return lower_keys.get(user_choice)`

Reply:
That is much easier to read, I'll be sure to implement this if this function is still needed when I implement the numbering for the menu system.